CODE-GUIDELINES.md

```diff
@@ -37,3 +37,59 @@ test('...', () => {
   vi.mocked(window.windowMethod).mockResolvedValue('a string');
 });
 ```
+
+### Mock before tests, not during tests
+
+When you mock functions for some tests, it is recommended to define the mocks from the `beforeEach` callback inside a `describe`,
+so the environment created by the mocks is clearly defined in the `describe`, and the `test` is only executing the tests:
+
+You also need to reset (or restore) the mocks from the `afterEach` callback, so these mocks are not used for other tests.
+
+Do not write:
+
+```ts
+vi.mock('node:fs');
+
+test('first behaviour', () => {
+  vi.mocked(fs.existsSync).mockReturnValue(true);
+  codeToTest();
+  expect(...);
+});
+
+test('second behaviour', () => {
+  vi.mocked(fs.existsSync).mockReturnValue(true);
+  otherCodeToTest();
+  expect(...);
+});
+```
+
+Instead, write:
+
+```ts
+import { existsSync, PathLike } from 'node:fs';
+import { afterEach, beforeEach, describe, expect, MockedFunction, test, vi } from 'vitest';
+
+vi.mock('node:fs');
+
+describe('the file exists', () => {
+  let existsSyncMock: MockedFunction<(path: PathLike) => boolean>;
```

Comment:
We should not try to declare the mock, we should access them by doing `vi.mocked()` or directly `expect(existsSync)`